test/e2e/openliberty_sso.go

```diff
@@ -0,0 +1,364 @@
+func testSocialLogin(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) error {
+	ns, err := ctx.GetNamespace()
+	if err != nil {
+		return err
+	}
+
+	const name string = "openliberty-sso"
+
+	// Create Secret for Github Login
+	secretTarget := types.NamespacedName{Name: name+"-olapp-sso", Namespace: ns}
+	data := map[string][]byte{
+		"github-clientId": []byte("bW9vb29vb28="),
+		"github-clientSecret": []byte("dGhlbGF1Z2hpbmdjb3c="),
+		"oidc-clientId": []byte("bW9vb29vb28="),
+		"oidc-clientSecret": []byte("dGhlbGF1Z2hpbmdjb3c="),
+	}
+
+	err = util.CreateSecretForSSO(f, ctx, secretTarget, data)
+	if err != nil {
+		return err
+	}
+	// Set up certificates for route, see: cert-manager test in RC
+	// terminationPolicy := v1.TLSTerminationReencrypt
+	expose := true
+	clusterIp := corev1.ServiceTypeClusterIP
+	githubLogin := v1beta1.GithubLogin{Hostname: "github.com"}
+	openliberty := util.MakeBasicOpenLibertyApplication(t, f, name, ns, 1)
+	openliberty.Spec.ApplicationImage = "openliberty/open-liberty:full-java8-openj9-ubi"
+	// TODO debug why these environment variables are causing it to crash loop
+	openliberty.Spec.Env = []corev1.EnvVar{
+		{Name: "SEC_TLS_TRUSTDEFAULTCERTS", Value: "true"},
+		{Name: "SEC_IMPORTS_K8S_CERTS", Value: "true"},
+	}
+	openliberty.Spec.Service = &v1beta1.OpenLibertyApplicationService{
+		Type: &clusterIp,
+		Port: 9080,
+		Certificate: &v1beta1.Certificate{},
+	}
+	openliberty.Spec.Expose = &expose
+	openliberty.Spec.SSO = &v1beta1.OpenLibertyApplicationSSO{
+		Github: &githubLogin,
+	}
+	err = f.Client.Create(goctx.TODO(), openliberty, &framework.CleanupOptions{TestContext: ctx, RetryInterval: time.Second, Timeout: time.Second })
+	if err != nil {
+		return err
+	}
+
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, ns, name, 1, retryInterval, timeout)
+	if err != nil {
+		return err
+	}
+
+	t.Log("verifying environment variables of containers")
+
+	target := types.NamespacedName{Name: name, Namespace: ns}
+	if err = verifyEnvVariables(ctx, f, target, secretTarget); err != nil {
+		return err
+	}
+	t.Log("secret data applied successfully")
+
+	// TODO: Update secret and verify values
+	secret := corev1.Secret{}
+	err = f.Client.Get(goctx.TODO(), secretTarget, &secret)
+	if err != nil {
+		return err
+	}
+
+	secret.Data = map[string][]byte{
+		"github-clientId": []byte("different="),
+		"github-clientSecret": []byte("differentSecret="),
+		"oidc-clientId": []byte("bW9vb29vb28="),
+		"oidc-clientSecret": []byte("dGhlbGF1Z2hpbmdjb3c="),
+		"twitter-clientId": []byte("twitterID="),
+		"twitter-clientSecret": []byte("twitterSecret"),
+	}
+	//TODO: why is twitter not being picked up?
+
+	err = f.Client.Update(goctx.TODO(), &secret)
+	if err != nil {
+		return err
+	}
+
+	err = util.WaitForPodUpdates(t, f, ctx, target, 1)
+	if err != nil {
+		return err
+	}
+
+	if err = verifyEnvVariables(ctx, f, target, secretTarget); err != nil {
+		return err
+	}
+	t.Log("secret data updated successfully")
+
```

Comment:
It'll be good to cover the scenario when the only update to secret is updating the value of an existing key (no new keys). It was an edge case for which we had to add some extra code (to force Deployment to pass the value to Pod).